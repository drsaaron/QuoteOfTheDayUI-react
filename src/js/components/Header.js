/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

import { goHome } from '../actions/NavigationActions';
import { connect } from 'react-redux';

const {version} = require("../../../package.json");

const mapStateToProps = (state) => {
    return {

    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        goHome: (navigate) => dispatch(goHome(navigate))
    };
};

const Header = (props) => {

    const handleHomeLink = (event) => {
        event.preventDefault();
        props.goHome(props.navigate);
    }

    return (
        <div id="pageHeader">
            <div id="pageHeaderApp">
                <a href="/" onClick={handleHomeLink}>Scott's Quote of the Day, version {version}</a>
            </div>
        </div>
    );
}

export default connect(mapStateToProps, mapDispatchToProps)(Header);
