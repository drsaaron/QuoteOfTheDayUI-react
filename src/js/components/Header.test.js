import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { createStore } from 'redux';

const rootReducer = (history) => { return {}; };
const store = createStore(rootReducer);

import Header from './Header';

describe('header', () => {
    test('basic', () => {
	const component = renderer.create(<Provider store={store}><Header /></Provider>);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
    });
});
