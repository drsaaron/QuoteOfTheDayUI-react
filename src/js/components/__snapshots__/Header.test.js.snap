// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header basic 1`] = `
<div
  id="pageHeader"
>
  <div
    id="pageHeaderApp"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Scott's Quote of the Day, version 
      1.2.3
    </a>
  </div>
</div>
`;
